source/MapPanel.cpp

```diff
@@ -102,14 +102,18 @@ void MapPanel::Draw()
 	if(!distance.HasRoute(selectedSystem))
 	{
 		const Font &font = FontSet::Get(18);
-		
-		static const string NO_ROUTE = "You have not yet mapped a route to this system.";
 		Color black(0., 1.);
 		Color red(1., 0., 0., 1.);
-		Point point(-font.Width(NO_ROUTE) / 2, Screen::Top() + 40);
 		
-		font.Draw(NO_ROUTE, point + Point(1, 1), black);
-		font.Draw(NO_ROUTE, point, red);
+		string errormsg = "You have no";
+		if(player.HasSeen(selectedSystem))
```

Comment:
HasSeen() returns true any time you know the system's name (e.g. because it's a mission destination), so this will lead to confusing messages in cases where the player has not actually mapped a route yet.

Also, splitting the string in the middle of a word is rather confusing, and `errormsg` doesn't follow the coding conventions for variable names (camelCase and avoiding abbreviations).